src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -364,12 +364,16 @@ private float addRow(
           throw new SFException(ErrorCode.INTERNAL_ERROR, "SubColumnFinder is not initialized.");
         }
 
-        for (String subColumn : subColumnFinder.getSubColumns(columnName)) {
-          RowBufferStats stats = statsMap.get(subColumn);
+        for (String subColumnId :
+            subColumnFinder.getSubColumns(fieldIndex.get(columnName).type.getId())) {
```

Comment:
fieldIndex.get(columnName) can return null, lets handle that and throw explicitly? OK in your next PR.

Reply:
The `columnName` is from `Set.difference(fieldIndex.keySet(), otherSet)`. I think this should be safe.